src/Clash/Cores/Ethernet/Examples/FullUdpStack.hs

```diff
@@ -0,0 +1,282 @@
+   This is then passed to the `icmpUdpStack`, which is preceded and followed by a packet buffer (`packetBufferC`). a circuit which stores words in a buffer until the packet is complete once a packet is complete it will send the entire packet out at once without stalls. If a word in a packet has `_abort` set to true, the packetBuffer will drop the entire packet.  
+   These packet buffers have a depth of 2^10 data bits and a depth of 2^4 meta bits. Since one packet may consist of many fragments, which by definition must have the same metadata, but have different content, we buffer metadata separately from content.
+   The reason for using packet buffers is because they absorb some backpressure generated by the @ipTx port  and the icmpUdpStack. In particular, the `toEthernetStreamC` asserts backpressure until arpC has given an `ArpResponse` to its `ArpLookup`.
+   From the packetbuffer, the stream is passed to `ipLitePacketizerC`, which is another convenience composition of `fromLiteC` and `ipPacketizerC`. `fromLiteC` simply maps the `_meta` field of the packet stream from an `IPv4HeaderLite` to an `IPv4Header`, while `ipPacketizer` packetizes the `IPv4Header` into the data of the packet stream, and puts the destination IP address into the `_meta` field. 
+   This is convenient since the following ARP lookup service can take the required IP address directly from the `_meta`.
+3. The `toEthernetStreamC` performs the ARP lookup and, if it gets an `ArpEntryFound` from `arpC` (in the backward line of the arplookup port), it constructs an `EthernetHeader` with the source and target MAC fields set to our MAC address and the found MAC address respectively, and the appropriate ethertype for IPv4.
+4. Finally, a `packetArbiterC` running in `RoundRobin` mode switches in a round-robin fashion between outputting a packet from the @arpEthOut stream and the @ipEthOut stream, i.e.
+   it alternatingly will output a complete packet coming from `toEthernetC` and a complete packet from `icmpUdpPacket`.
+
+All code thus far described handles interfacing between the MAC and IP layer. `icmpUdpStack` handles the payload of the IP packets, implementing two functionalities:
+ - sending an appropriate ICMP echo response to an ICMP echo request. 
+ - depacketizing, handling and packetizing UDP traffic with the provided UPD handler circuit.
+`icmpUdpStack`, line by line, does the following:
+
+1. The `packetDispatcher` + `routeBy` combination is now used to splite the input stream of type `PacketStream dom 4 IPv4HeaderLite` based on the protocol in the payload. The protocol number in the IP v4 header is also present iin `IPv4HeaderLite`, namely 
+   the field `_ipv4lProtocol`. 
+2. The stream `icmpIn` of ICMP packets is passed to `icmpEchoResponderC`. This circuit takes in a `Signal IPv4Adress` namely our IP, to internally
+  - parses the ICMP header from the content of the packet using the generic depacketizer and puts this into the metadata in a tuple with the 
```

Comment:
I'm not sure if haddock can do nested lists. In either case, it's not correctly formatted now

Reply:
doc says it can, namely with 

* this is a bullet point

** this is nested

But this doesn't seem to work with multiline comments. Another way of doing it would be (according to the docs)

* this is a bullet point
     * four spaces makes an indented bullet point.
also doesn't work! I give up.